### PR TITLE
chore(blog): add recovery key blog

### PR DESCRIPTION
**Pull request by vbudhram**
Mirror of https://docs.google.com/document/d/1cyiSidVFK6iR3jhD8U8UYU3aI1ZjEukCbaN1V_jWVbk/edit#.

Don't merge until released to 10% of users (train 121).

### CODE REVIEW

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+The Firefox Accounts (FxA) team is in the process of releasing a
+new feature called Account Recovery. This feature is aimed at
+solving one of the most frustrating issues for our users. When
+a user resets their account password, they risk losing data such
+as synced bookmarks, passwords and browsing history. FxA encrypts
+synced browser data with the user’s account password in order to
+protect their privacy. Resetting the password destroys the
+encryption key and leaves the data inaccessible which
+eventually be deleted.
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: `eventually be deleted` => `eventually is deleted`

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+the updated encryption keys.
+
+The security requirements for the feature:
+
+1. The recovery key must be short enough that it can be reliably stored and retrieved by the user without corruption on a printout.
+2. The recovery key must be long enough that brute-force guessing a token is infeasible (online or offline)
+3. The combination of the recovery key plus some information held by the FxA server must be sufficient to recover kB.
+4. It must be infeasible to recover kB from the recovery key alone, without retrieving additional data from the Firefox Account server. (This is a safeguard against leakage of the token from wherever the user stores it).
+5. It must be infeasible to recover kB from data held by the Firefox Account server alone, without also possessing the recovery key.  (This is a safeguard against Mozilla, and leaks of our server-side database state).
+6. It must be possible for the holder of the recovery key to convince the Firefox Account server that they hold a valid key, without revealing the key itself. (This is a safeguard Mozilla, and leaks of our server-side database state.)
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: `a safeguard Mozilla` => `a safeguard against Mozilla` I think?

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+* Unique user ID (uid)
+  * User’s unique account identifier
+
+Our servers will ever only know and interact with the recovery
+key ID. By doing this, FxA safeguards itself from ever knowing
+the raw recovery key and therefore would not be able to derive
+kB to decrypt any data. Even if there was a database breach,
+an attacker would still not know the original recovery key.
+
+Using the values above, a JSON Web Key [(JWK)](https://en.wikipedia.org/wiki/JSON_Web_Token) is created with the following properties.
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: Do you want to link to a JWT article here? The [JWK spec](https://tools.ietf.org/html/rfc7517) is a bit dry but pretty approachable as far as specs go, and the [introduction](https://tools.ietf.org/html/rfc7517#section-1) has most of the applicable info.

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+Below is a diagram that illustrates where the above steps happen:
+
+![](/application-services/img/blog/2018-09-10/setup-recovery-key-diagram.png)
+
+### Consuming the recovery key
+
+1. From login page, the user initiates password reset and clicks `reset password` from email received
+  * After clicking the link in the email, a temporary session token is created for subsequent requests
+2. User retrieves recovery key from where they saved it. Ex printout
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: `Ex printout` => `For example, a printout`

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+3. User enters their recovery key into password reset form.
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: Some of these list items end with a period, others do not.

File: website/blog/2018-09-10-account-recovery-in-firefox-accounts.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+4. Client generates JWK from recovery key
+5. Client requests for recovery data using the recovery key id
+6. Client decrypts the recovery data using the derived encryption key to obtain the kB
```
Comment (shane-tomlinson) [inline website/blog/2018-09-10-account-recovery-in-firefox-accounts.md]: `derived encryption key` is the JWK from step 4?

**Comment by vladikoff** [comment]
121 is live!

**Pull request merged**